lib/elixir/lib/enum.ex

```diff
@@ -932,7 +932,8 @@ defmodule Enum do
 
   @doc """
   Returns a new enumerable appending the result of invoking `fun` on
-  each corresponding item of `enumerable`.
+  each corresponding item of `enumerable` and flattening
+  the given results (only one level deep)..
```

Comment:
@PatNowak let's rephrase the whole documentation for this function then. What about something like this:

```elixir
@doc """
Maps the given `fun` over `enumerable` and flattens the result.

This function returns a new enumerable built by appending the result of invoking `fun`
on each element of `enumerable` together; conceptually, this is similar to a
combination of `map/3` and `concat/1`.
"""
```

Wdyt? This way the summary (first line) is short and to the point, and the description is a bit more helpful.

Btw, if we're gonna make this change we need to make it to the documentation for `Stream.flat_map/2` as well. :)